### DISCUSSION TITLE
Marketplace example help.

### DISCUSSION THREAD

**Discussion by jbeckton** [medusajs/medusa#9223] [Q&A] [answered]
I cloned the marketplace example from the repo. Ran it locally and added an admin user via the cli. I can log into the local instance so everything seems to be functioning properly.

The part I am struggling with is there is no documentation in the repo regarding what to do next. I downloaded the yml file for Postman, hit some of the endpoints but I am at a loss as to what these endpoints do.

is there any docs out there that explain what these added endpoints do? How do I create a vendor and vendor login with them?

http://localhost:9000/auth/vendor/emailpass 
sent it an email and password and got a token back.
{
    "email":"vendor2@example.com",
    "password":"password"
}

http://localhost:9000/auth/vendor/emailpass/register
sent it an email and password and got a token back.
{
    "email":"vendor2@example.com",
    "password":"password"
}

http://localhost:9000/vendors

sent:

{
    "name": "Vendor Two",
    "handle": "Vend2",
    "admin": {
        "email":"vendor2@example.com",
        "first_name": "Vendor2",
        "last_name": "Admin"
    }
    
}

got back a large vendor object

http://localhost:9000/vendors/products
use token from auth response and get back

{
    "message": "Unauthorized"
}

**Comment by jbeckton** [comment, marked as answer]
found the answer at the bottom of this page...

https://docs.medusajs.com/v2/resources/recipes/marketplace/examples/vendors

**Comment by alisafavi** [comment]
Hi, how can create a page for login for venders

**Reply by blbergo** [reply]
After 4 days of struggling, I found a work around based on [this tutorial article](https://medium.com/@igorkhomenko/building-a-multivendor-marketplace-with-medusa-js-2-0-a-dev-guide-f55aec971126) by Igor Khomenko. Then add this to your create-vendor-admin workflow, add a password field to the API route, zod schema and workflow.

src/workflows/marketplace/create-vendor-admin/steps/create-user.ts:
```ts
import { Modules } from "@medusajs/framework/utils"
import { createStep, StepResponse } from "@medusajs/framework/workflows-sdk"
import { CreateVendorAdminWorkflowInput } from ".."

const createUserStep = createStep(
  "create-user",
  async (
    { admin, authIdentityId }: CreateVendorAdminWorkflowInput,
    { container }
  ) => {
    const userModuleService = container.resolve(Modules.USER)
    const authModuleService = container.resolve(Modules.AUTH)

    const user = await userModuleService.createUsers(admin)

    await authModuleService.updateAuthIdentities({
      id: authIdentityId,
      app_metadata: {
        user_id: user.id,
      },
    })

    return new StepResponse({ user }, user.id)
  },
  async (id: string, { container }) => {
    const userModuleService = container.resolve(Modules.USER)

    await userModuleService.deleteUsers([id])
  }
)

export default createUserStep
```

src/workflows/marketplace/index.ts:
```ts
import {
  createWorkflow,
  WorkflowResponse,
} from "@medusajs/framework/workflows-sdk"
import { setAuthAppMetadataStep } from "@medusajs/medusa/core-flows"
import createVendorAdminStep from "./steps/create-vendor-admin"
import createUserStep from "./steps/create-user"

export type CreateVendorAdminWorkflowInput = {
  admin: {
    email: string
    password: string
    first_name?: string
    last_name?: string
    vendor_id: string
  }
  authIdentityId: string
}

const createVendorAdminWorkflow = createWorkflow(
  "create-vendor-admin",
  function (input: CreateVendorAdminWorkflowInput) {
    const vendorAdmin = createVendorAdminStep({
      admin: input.admin,
    })

    createUserStep(input)

    setAuthAppMetadataStep({
      authIdentityId: input.authIdentityId,
      actorType: "vendor",
      value: vendorAdmin.id,
    })

    return new WorkflowResponse(vendorAdmin)
  }
)

export default createVendorAdminWorkflow
```
